src/selector.js

```diff
@@ -0,0 +1,39 @@
+export const PATH_KEY = Symbol.for('@i18next/PATH_KEY');
+
+function join(path, opts) {
+  return path.reduce((acc, cur) => {
+    if (acc.endsWith(opts.nsSeparator)) {
+      return `${acc.slice(0, -opts.nsSeparator.length)}${opts.nsSeparator}${cur}`;
+    }
+    if (acc.length === 0) {
+      return `${cur}`;
+    }
+    return `${acc}.${cur}`;
```

Comment:
Like `nsSeparator` there is also a `keySeparator`

https://www.i18next.com/overview/configuration-options#others

Reply:
Good call. Removed in 2bf99f3686adaea62ffdf58a4eeba6f018aae466